ContactsManagerBackend/ContactsManager/ContactsManagerWebApi/Controllers/ContactsController.cs

```diff
@@ -10,7 +14,65 @@ public class ContactsController : ApiController
         // GET: Contacts
         public IHttpActionResult Get()
         {
-            return Ok(ContactsManager.GetContacts());
+            return Ok<List<Contact>>(Mapper.Map<List<Contact>>(ContactsManager.GetContacts()));
+        }
+
+        public IHttpActionResult Get(Guid Id)
+        {
+            var contact = ContactsManager.GetContactById(Id);
+
+            if(contact == null)
+            {
+                return NotFound();
+            }
+            else
+            {
+                return Ok<Contact>(Mapper.Map<Contact>(contact));
+            }
+        }
+
+        [HttpPost]
+        public IHttpActionResult Post([FromBody] Contact contact)
+        {
+            if (
+                String.IsNullOrEmpty(contact.FirstName) ||
+                String.IsNullOrEmpty(contact.LastName) ||
+                String.IsNullOrEmpty(contact.Email) ||
+                String.IsNullOrEmpty(contact.Birthdate.ToString())
```

Comment:
This is meaningless. Currently the property is not nullable, so it can never be empty or null.
When you change is to nullable, you should check for null rather than convert to string.